fem/fe/fe_base.hpp

```diff
@@ -789,7 +789,6 @@ class VectorFiniteElement : public FiniteElement
 protected:
    bool is_nodal;
 #ifndef MFEM_THREAD_SAFE
-   mutable DenseMatrix J, Jinv;
```

Comment:
This feels wrong... don't we need these in the class for thread safety?

Reply:
I still believe my previous comment regarding `Jinv`, however, you're probably correct about `J`. This is used as a local variable to store the value of `J^T * J` so I was probably wrong to remove it.

I've put `J` back but renamed it `JtJ`. It seems to be used in only one place, `ProjectCulr_ND`, where I've used the usual pattern of declaring a new variable or resizing a member variable based on the thread safety flag.